src/utils.ts

```diff
@@ -5,6 +5,14 @@ import type {ReactVideoSource, ReactVideoSourceProperties} from './types/video';
 
 type Source = ImageSourcePropType | ReactVideoSource;
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function generateHeaderForNative(obj?: Record<string, any>) {
+  if (!obj) {
+    return [];
```

Comment:
Are you sure about this change ?
native code is made to handle an object, not an array ...

Reply:
Ah, It's because of codegen issue.
Codegen cannot parse `Record<string, string>` type.
So that's why I made this util.
And through this utils function, it converts `headers` as array.

Reply:
No problem with this internal refactor!
I saw you changed parsing on android, I think you should also do it on ios ?

Reply:
@freeboub I handled iOS data structure to this commit https://github.com/react-native-video/react-native-video/pull/3487/commits/6c0fe6885fcbeca107174fe3040ee4812eab74b9
Can you have a check please?